source/slang/slang.cpp

```diff
@@ -808,18 +808,40 @@ SLANG_API char const* spGetTranslationUnitSource(
     int                     translationUnitIndex)
 {
     auto req = REQ(request);
-    return req->translationUnits[translationUnitIndex]->result.outputSource.Buffer();
+    return (char const*)req->translationUnits[translationUnitIndex]->result.outputSource.Buffer();
+}
+
+SLANG_API void const* spGetTranslationUnitCode(
+    SlangCompileRequest*    request,
+    int                     translationUnitIndex,
+    size_t*                 outSize)
+{
+    auto req = REQ(request);
+    Slang::TranslationUnitResult& result = req->translationUnits[translationUnitIndex]->result;
+    *outSize = (size_t)result.outputSource.Count();
```

Comment:
I've already updated it to use the per-entry-point interface in the branch that I'm trying to get Nir to merge.
There is no way we can use the whole-translation-unit interface with binary formats anyway, so no matter what we'll need to move to the per-entry-point approach as part of this shift.